### ISSUE TITLE
Improve documentation - add warning in section `Option: purge_statistics` for users who use Energy dashboard

### ISSUE THREAD

**Issue by Arteusz** [lmagyar/homeassistant-addon-mariadb-inmemory#58]
First Thanks for your work, had to replace sd card every year before this add-on.

Please consider adding additional warning in section `Option: purge_statistics` for users that use Energy dashboard.

There is weird behavior in Energy Dashboard when statistic have been purged. Energy usage graph pads values for the first month it has data for. 

In attached screenshot:
- earliest available statistics are from march  ( purge_statistics,  period: monthly,  keep_periods: 6)
- yearly energy consumption values are correct 
- value for march is padded - should be around 200 
- values for later months are correct

![image](https://github.com/lmagyar/homeassistant-addon-mariadb-inmemory/assets/2125455/598b0e64-437b-446f-8da5-58554a690d1b)

**Comment by Arteusz** [comment]
Modification of purging code, to delete from 00:00h local time is a great idea but it's not the source of the problem. 

Statistics for march have correct values in the database so the padding is "UI only".  It is worth mentioning that energy used statistic (at least in my case) is stored as increasing value

![image](https://github.com/lmagyar/homeassistant-addon-mariadb-inmemory/assets/2125455/bb97ce61-5dc8-48a6-b2eb-b978b05b5953)


I found the code responsible for calculating data for the chart [hui-energy-devices-graph-card.ts](https://github.com/home-assistant/frontend/blob/4b5c7021ff06bb175efe94d42bc2b07e04126628/src/panels/lovelace/cards/energy/hui-energy-devices-graph-card.ts#L248C67-L248C67). From a quick glance it looks like values returned, for the chart, are returned as change between statistic records. So the data look like this:

| Month | energy statistic value | change to previous value |
|---------|:----------------------:|:--------:|
| Jan | 0 (purged) | 0 |
| Feb | 0 (purged) | 0 |
| Mar | 800 | 800 |
| Apr | 1000 | 200 |
| May | 1200 | 200 | 

That's why the first month that has statistic gets padded. It's not an error per se but just a quirky behavior of energy usage graph. 

As for the restoring of statistic, I had a backup from April that contained the statistic before purge, so I just extracted the data  and imported it to HA database. After restoring everything looks good. 

Suggested warning:
> ---
>
> **Warning**
>
> ---
>
> Purging of statistic impacts `Energy dashboard - Energy usage graph` in a weird way. Reported usage for the first period after purge gets padded.
> <details>
>
> Values for the bar plot of `Energy dashboard - Energy used graph` are calculated as change between statistic values. If you purge statistics, the first reported value on energy used graph will be padded. 
>
> Example - values before March were purged:
>
> | Month | energy statistic value <br> (before purge) | change to previous value <br> (before purge) | energy statistic value <br> (after purge) | change to previous value <br> (after purge) |
> |---------|:----------------------:|:--------:|:----:|:----:|
> | Jan | 400 | - | 0 | 0 |
> | Feb | 600 | 200 | 0 | 0 |
> | Mar | 800 | 200 | 800 | 800 |
> | Apr | 1000 | 200 | 1000 | 200 |
> | May | 1200 | 200 | 1200 | 200|
> </details>


Raw markdown:
-
``````
> ---
>
> **Warning**
>
> ---
>
> Purging of statistic impacts `Energy dashboard - Energy usage graph` in a weird way. Reported usage for the first period after purge gets padded.
> <details>
>
> Values for the bar plot of `Energy dashboard - Energy used graph` are calculated as change between statistic values. If you purge statistics, the first reported value on energy used graph will be padded. 
>
> Example - values before March were purged:
>
> | Month | energy statistic value <br> (before purge) | change to previous value <br> (before purge) | energy statistic value <br> (after purge) | change to previous value <br> (after purge) |
> |:---:|:-----:|:---:|:----:|:---:|
> | Jan | 400   | -   | 0    | 0   |
> | Feb | 600   | 200 | 0    | 0   |
> | Mar | 800   | 200 | 800  | 800 |
> | Apr | 1000  | 200 | 1000 | 200 |
> | May | 1200  | 200 | 1200 | 200 |
> </details>

**Comment by lmagyar** [comment]
Thank you for your investigation! Without energy data in my systems, I wouldn't be able to figure it out!

I've added a bit shorter version of the warning to the documentation and referenced this issue for your images. After HA refreshes the repository/update information of the add-ons, it will be available without releasing a new version.

---

I've tried to figure out the exact UTC time for local midnight X month ago even if it is through a daylight saving change, it requires coreutils' `date` and some black magic, I won't add it to the code, it won't solve any problem currently, just document it here.

```
date -d "$(date -d '5 months ago' '+%Y-%m-%d 00:00:00 %z')" -u '+%Y-%m-%d %H:%M:%S'
2023-04-21 22:00:00

date -d "$(date -d '6 months ago' '+%Y-%m-%d 00:00:00 %z')" -u '+%Y-%m-%d %H:%M:%S'
2023-03-21 23:00:00
```
